### PR TITLE
APIS-4825 Add ApiVersion Value Class

### PR DESCRIPTION
**Pull request by anjumabbas5**

### CODE REVIEW

File: app/connectors/ApplicationConnector.scala [hunk 1]
```diff
@@ -244,10 +244,10 @@ abstract class ApplicationConnector(implicit val ec: ExecutionContext) extends R
     encode(str, encoding)
   }
 
-  def searchCollaborators(apiContext: ApiContext, apiVersion: String, partialEmailMatch: Option[String])(implicit hc: HeaderCarrier): Future[Seq[String]] = {
+  def searchCollaborators(apiContext: ApiContext, apiVersion: ApiVersion, partialEmailMatch: Option[String])(implicit hc: HeaderCarrier): Future[Seq[String]] = {
     val queryParameters = List(
       "context" -> apiContext.value,
-      "version" -> apiVersion
+      "version" -> apiVersion.value
```
Comment (AndySpaven) [inline app/connectors/ApplicationConnector.scala:250]: I wonder if these need urlEncoding.

File: app/utils/UserFunctionsWrapper.scala [hunk 2]
```diff
@@ -42,7 +42,7 @@ trait UserFunctionsWrapper {
     def toKeyValue(api: APIDefinition, version: ApiVersionDefinition) = {
```
Comment (AndySpaven) [inline app/utils/UserFunctionsWrapper.scala:42]: Can we rename this as versionDefn to disambiguate and remove the version.version mentions

File: app/connectors/ApplicationConnector.scala (hunk 1 above)
Comment (anjumabbas5) [reply app/connectors/ApplicationConnector.scala:250]: They weren't urlEncoded before so I left as was

**Review by AndySpaven — APPROVED** [review]

File: app/connectors/ApplicationConnector.scala (hunk 1 above)
Comment (AndySpaven) [reply app/connectors/ApplicationConnector.scala:250]: That means nothing ;-)

**Pull request merged**